### PR TITLE
feat: Custom log and error handlers

### PR DESCRIPTION
**Pull request by DhansAL**
# Description

 This closes issue #141 and adds some code enhancements.

what this pull request includes :

1.  custom log and error handlers which include additional features like, 

-  logging according to the severity of bug or log ( 7 different levels )
-  the capability to export all the logs of different levels to an external log file while production
-  control over console format (include timestamps etc )

2.  Added error handling on all modules.
3.  Restructured the code, added src directory.

## Check List (mark all applicable checkboxes) 

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [ ] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

## Screenshots (if applicable)

 Original           | Updated
 :--------------------: |:--------------------:
 original screenshot | updated screenshot |

### CODE REVIEW

File: src/Commands/DSA.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
+const { botErrorHandler } = require('../utils/botErrorHandler');
+
+try {
+	module.exports = {
+		name: 'dsa',
+		description: 'sends the link for the dsa course',
```
Comment (Panquesito7) [inline src/Commands/DSA.js:6]:
```suggestion
		description: 'Sends the link and information for the DSA course',
```

File: src/Commands/DSA.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+						name: 'Complete Java + DSA course',
+						value:
+							'https://www.youtube.com/playlist?list=PL9gnSGHSqcnr_DxHsP7AW9ftq0AtAyYqJ',
+					},
+					{
+						name: 'Code sample and full syllabus for DSA lectures',
+						value: 'https://github.com/kunal-kushwaha/DSA-Bootcamp-Java',
+					},
+					{
+						name: 'Try hands on git',
```
Comment (Panquesito7) [inline src/Commands/DSA.js:29]:
```suggestion
						name: 'Try hands on Git',
```

File: src/Commands/help.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+	module.exports = {
+		name: 'help',
+		description: 'sends the help command',
+		execute(message, args, Discord) {
+			const helpEmbed = new Discord.MessageEmbed()
+				.setColor('#2e97c2')
+				.setTitle('Community monitor for help')
+				.setURL('https://github.com/commclassroom/classroom-monitor-bot')
+				.setAuthor(
+					'Classroon monitor',
```
Comment (Panquesito7) [inline src/Commands/help.js:13]:
```suggestion
					'Classroom monitor',
```

File: src/Commands/links.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+	module.exports = {
+		name: 'links',
+		description: 'sends the link relevant to comm classroom',
+		execute(message, args, Discord) {
+			const linkEmbed = new Discord.MessageEmbed()
+				.setColor('#2e97c2')
+				.setTitle('Community Classroom Social accounts')
+				.setURL('https://github.com/commclassroom/classroom-monitor-bot')
+				.setAuthor(
+					'Classroon monitor',
```
Comment (Panquesito7) [inline src/Commands/links.js:13]:
```suggestion
					'Classroom monitor',
```

File: src/Commands/DSA.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+	module.exports = {
+		name: 'dsa',
+		description: 'sends the link for the dsa course',
+		execute(message, args, Discord) {
+			const linkEmbed = new Discord.MessageEmbed()
+				.setColor('#2e97c2')
+				.setTitle('Community Classroom DSA course details')
+				.setURL('https://github.com/commclassroom/classroom-monitor-bot')
+				.setAuthor(
+					'Classroon monitor',
```
Comment (Panquesito7) [inline src/Commands/DSA.js:13]:
```suggestion
					'Classroom monitor',
```

File: src/Commands/links.js [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+						name: 'Instagram',
+						value: 'https://www.instagram.com/commclassroom/',
+					},
+					{
+						name: 'Youtube',
+						value: 'https://www.youtube.com/c/KunalKushwaha/',
+					},
+					{
+						name: 'Discord Invite',
+						value: 'https://discord.gg/K9kxUXvfND',
```
Comment (Panquesito7) [inline src/Commands/links.js:37]:
```suggestion
						value: 'https://discord.gg/commclassroom',
```

File: src/Commands/links.js [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+					{
+						name: 'Twitter',
+						value: 'https://twitter.com/commclassroom/',
+					},
+					{
+						name: 'Instagram',
+						value: 'https://www.instagram.com/commclassroom/',
+					},
+					{
+						name: 'Youtube',
```
Comment (Panquesito7) [inline src/Commands/links.js:32]:
```suggestion
						name: 'YouTube',
```

File: src/Commands/presence.js [hunk 8]
```diff
@@ -0,0 +1,15 @@
+const { botErrorHandler } = require('../utils/botErrorHandler');
+try {
+	module.exports = async function (client) {
+		client.user.setPresence({
+			status: 'online',
+			activity: {
+				name: 'cm!help',
+				type: 'LISTENING',
+				url: 'https://github.com/commclassroom/classroom-monitor-bot.',
```
Comment (Panquesito7) [inline src/Commands/presence.js:9]: I think this dot should be removed.
```suggestion
				url: 'https://github.com/commclassroom/classroom-monitor-bot',
```

File: src/Commands/source.js [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+	module.exports = {
+		name: 'source',
+		description: 'sends the source code of the bot',
+		execute(message, args, Discord) {
+			const linkEmbed = new Discord.MessageEmbed()
+				.setColor('#2e97c2')
+				.setTitle('Community Classroom Discord bot')
+				.setURL('https://github.com/commclassroom/classroom-monitor-bot')
+				.setAuthor(
+					'Classroon monitor',
```
Comment (Panquesito7) [inline src/Commands/source.js:12]:
```suggestion
					'Classroom monitor',
```

File: src/Commands/source.js [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+					'https://i.imgur.com/yMCOBLH.png',
+					'https://discord.js.org'
+				)
+				.setDescription(
+					'Open source discord bot made for ✨Community Classroom✨'
```
Comment (Panquesito7) [inline src/Commands/source.js:17]:
```suggestion
					'Open Source Discord bot made by and for ✨Community Classroom✨'
```

File: src/Commands/source.js [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+				)
+				.addFields({
+					name: 'Github',
```
Comment (Panquesito7) [inline src/Commands/source.js:20]:
```suggestion
					name: 'GitHub',
```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Awesome work! 🚀

**Comment by DhansAL** [comment]
done ^_^

**Comment by DhansAL** [comment]
if there's a problem with my pull request? please let me know  ^_^

File: src/Commands/links.js [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+					{
+						name: 'Discord Invite',
+						value: 'https://discord.gg/commclsassroom',
+					},
+					{
+						name: 'Telegram Invite',
+						value: 'https://t.me/commclassroom',
+					},
+					{
+						name: 'Become University Leader',
```
Comment (SiddhantKhisty) [inline src/Commands/links.js:44]: University Leader Program was discontinued. Kindly Remove this field

**Review by SiddhantKhisty — COMMENTED** [review]
Looks good except for a few typos. Please fix the typos suggested by Panquesito. Also kindly remove the University Leader field as the program is discontinued

File: src/Commands/links.js (hunk 12 above)
Comment (DhansAL) [reply src/Commands/links.js:44]: sure!

**Comment by DhansAL** [comment]
I have removed all typos and the deprecated University field leader part. please have a look :)

**Review by SiddhantKhisty — APPROVED** [review]
LGTM

File: src/Commands/help.js (hunk 3 above)
Comment (Panquesito7) [inline src/Commands/help.js:13]:
```suggestion
					'Classroom Monitor',
```

File: src/Commands/help.js [hunk 13]
```diff
@@ -0,0 +1,32 @@
+const { botErrorHandler } = require('../utils/botErrorHandler');
+
+try {
+	module.exports = {
+		name: 'help',
+		description: 'sends the help command',
```
Comment (Panquesito7) [inline src/Commands/help.js:6]:
```suggestion
		description: 'Sends a list of valid commands (a.k.a. help command)',
```

File: src/Commands/links.js [hunk 14]
```diff
@@ -0,0 +1,54 @@
+const { botErrorHandler } = require('../utils/botErrorHandler');
+
+try {
+	module.exports = {
+		name: 'links',
+		description: 'sends the link relevant to comm classroom',
```
Comment (Panquesito7) [inline src/Commands/links.js:6]:
```suggestion
		description: 'Sends all Community Classroom's socials',
```

File: src/Commands/links.js [hunk 15]
```diff
@@ -0,0 +1,54 @@
...
+		execute(message, args, Discord) {
+			const linkEmbed = new Discord.MessageEmbed()
+				.setColor('#2e97c2')
+				.setTitle('Community Classroom Social accounts')
+				.setURL('https://github.com/commclassroom/classroom-monitor-bot')
+				.setAuthor(
+					'Classroon monitor',
```
Comment (Panquesito7) [inline src/Commands/links.js:13]:
```suggestion
					'Classroom Monitor',
```

File: src/Commands/source.js [hunk 16]
```diff
@@ -0,0 +1,33 @@
+const { botErrorHandler } = require('../utils/botErrorHandler');
+try {
+	module.exports = {
+		name: 'source',
+		description: 'sends the source code of the bot',
```
Comment (Panquesito7) [inline src/Commands/source.js:5]:
```suggestion
		description: 'Sends a link to the bot's source code',
```

File: src/Commands/source.js (hunk 9 above)
Comment (Panquesito7) [inline src/Commands/source.js:12]:
```suggestion
					'Classroom Monitor',
```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Many things are not fixed yet. Please fix them. 🙂

**Comment by DhansAL** [comment]
> Many things are not fixed yet. Please fix them. 🙂

please check now ^_^

File: src/Commands/DSA.js [hunk 17]
```diff
@@ -10,7 +10,11 @@ try {
 				.setTitle('Community Classroom DSA course details')
 				.setURL('https://github.com/commclassroom/classroom-monitor-bot')
 				.setAuthor(
+<<<<<<< HEAD
+					'Classroon Monitor',
+=======
 					'Classroom Monitor',
+>>>>>>> 291166d1ffd5a023d12a73d1868b7227364aa00d
```
Comment (Panquesito7) [inline src/Commands/DSA.js]:
```suggestion
					'Classroom Monitor',
```

**Comment by kaiwalyakoparkar** [comment]
@Panquesito7 Can you have a look at this again (As you have gone through once and suggested changes)

File: src/Commands/DSA.js [hunk 18]
```diff
@@ -0,0 +1,45 @@
...
+		description: 'Sends the link for the DSA course',
+		execute(message, args, Discord) {
+			const linkEmbed = new Discord.MessageEmbed()
+				.setColor('#2e97c2')
+				.setTitle('Community Classroom DSA course details')
+				.setURL('https://github.com/commclassroom/classroom-monitor-bot')
+				.setAuthor(
+					'Classroon Monitor',
+
+					'Classroom Monitor',
```
Comment (Panquesito7) [inline src/Commands/DSA.js:14]:
```suggestion
					'Classroom Monitor',
```

**Comment by DhansAL** [comment]
this merge took longer than usual @Panquesito7  thanks for reviewing and guiding through my errors ^_^

**Comment by DhansAL** [comment]
This pull request is covered by pull request #146 .

File: src/Commands/DSA.js [hunk 19]
```diff
@@ -0,0 +1,42 @@
...
+	module.exports = {
+		name: 'dsa',
+		description: 'Sends the link for the DSA course',
+		execute(message, args, Discord) {
+			const linkEmbed = new Discord.MessageEmbed()
+				.setColor('#2e97c2')
+				.setTitle('Community Classroom DSA course details')
+				.setURL('https://github.com/commclassroom/classroom-monitor-bot')
+				.setAuthor(
+					'Classroon Monitor',
```
Comment (Panquesito7) [inline src/Commands/DSA.js:12]:
```suggestion
					'Classroom Monitor',
```

File: src/Commands/DSA.js [hunk 20]
```diff
@@ -0,0 +1,42 @@
...
+						value: 'https://github.com/kunal-kushwaha/DSA-Bootcamp-Java',
+					},
+					{
+						name: 'Try hands on Git',
+						value: 'https://github.com/kunal-kushwaha/commclassroomOP',
+					}
+				)
+				.setTimestamp()
+				.setFooter(
+					'Want help? Classroom monitor is just `cm!help` far',
```
Comment (Panquesito7) [inline src/Commands/DSA.js:34]:
```suggestion
					'Want help? Classroom Monitor is just `cm!help` far',
```

File: src/Commands/help.js [hunk 21]
```diff
@@ -0,0 +1,32 @@
...
+	module.exports = {
+		name: 'help',
+		description: 'Sends a list of valid commands (a.k.a. help command)',
+		execute(message, args, Discord) {
+			const helpEmbed = new Discord.MessageEmbed()
+				.setColor('#2e97c2')
+				.setTitle('Community monitor for help')
+				.setURL('https://github.com/commclassroom/classroom-monitor-bot')
+				.setAuthor(
+					'Classroon Monitor',
```
Comment (Panquesito7) [inline src/Commands/help.js:13]:
```suggestion
					'Classroom Monitor',
```

File: src/Commands/help.js [hunk 22]
```diff
@@ -0,0 +1,32 @@
...
+					'https://discord.js.org'
+				)
+				.setDescription('Quality Education. Free For All. Forever.')
+				.addFields({
+					name: 'Current Supported Commands',
+					value: '`hey`, `help`,`version`,`links`,`translate`,`meme`,`source`',
+				})
+				.setTimestamp()
+				.setFooter(
+					'Want help? Classroom monitor is just `cm!help` far',
```
Comment (Panquesito7) [inline src/Commands/help.js:24]:
```suggestion
					'Want help? Classroom Monitor is just `cm!help` far',
```

File: src/Commands/links.js [hunk 23]
```diff
@@ -0,0 +1,54 @@
...
+	module.exports = {
+		name: 'links',
+		description: `Sends all Community Classroom's socials`,
+		execute(message, args, Discord) {
+			const linkEmbed = new Discord.MessageEmbed()
+				.setColor('#2e97c2')
+				.setTitle('Community Classroom Social accounts')
+				.setURL('https://github.com/commclassroom/classroom-monitor-bot')
+				.setAuthor(
+					'Classroon Monitor',
```
Comment (Panquesito7) [inline src/Commands/links.js:13]:
```suggestion
					'Classroom Monitor',
```

File: src/Commands/links.js [hunk 24]
```diff
@@ -0,0 +1,54 @@
...
+						value: 'https://discord.gg/commclsassroom',
+					},
+					{
+						name: 'Telegram Invite',
+						value: 'https://t.me/commclassroom',
+					}
+				)
+				.setTimestamp()
+				.setFooter(
+					'Want help? Classroom monitor is just `cm!help` far',
```
Comment (Panquesito7) [inline src/Commands/links.js:46]:
```suggestion
					'Want help? Classroom Monitor is just `cm!help` far',
```

File: src/Commands/source.js [hunk 25]
```diff
@@ -0,0 +1,34 @@
...
+		name: 'source',
+		description: `Sends a link to the bot's source code`,
+
+		execute(message, args, Discord) {
+			const linkEmbed = new Discord.MessageEmbed()
+				.setColor('#2e97c2')
+				.setTitle('Community Classroom Discord bot')
+				.setURL('https://github.com/commclassroom/classroom-monitor-bot')
+				.setAuthor(
+					'Classroon Monitor',
```
Comment (Panquesito7) [inline src/Commands/source.js:13]:
```suggestion
					'Classroom Monitor',
```

File: src/Commands/source.js [hunk 26]
```diff
@@ -0,0 +1,34 @@
...
+				.setDescription(
+					'Open Source Discord bot made by and for ✨Community Classroom✨'
+				)
+				.addFields({
+					name: 'GitHub',
+					value: 'https://github.com/commclassroom/classroom-monitor-bot',
+				})
+				.setTimestamp()
+				.setFooter(
+					'Want help? Classroom monitor is just `cm!help` far',
```
Comment (Panquesito7) [inline src/Commands/source.js:26]:
```suggestion
					'Want help? Classroom Monitor is just `cm!help` far',
```

**Review by Panquesito7 — APPROVED** [review]
These are all the issues that I see. After this, I'll give my approval. 🚀

**Comment by DhansAL** [comment]
done!

**Review by Panquesito7 — APPROVED** [review]
LGTM 🚀 Thank you for your contribution! 😄👍

**Pull request merged**